lib/Qandidate/Toggle/ExpressionCondition.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Qandidate\Toggle;
+
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+
+/**
+ * A condition written as a symfony language expression that gets evaluated against the
+ * full context, allowing access to several keys of the context in a single condition
+ */
+class ExpressionCondition extends Condition
+{
+    protected $expression;
+    protected $language;
+
+    /**
+     * @param string             $expression The expression to ve evaluated
+     * @param ExpressionLanguage $language The instance of the Expression Language
+     */
+    function __construct($expression, ExpressionLanguage $language = null)
+    {
+        $this->expression = $expression;
+        $this->language   = $language ?: new ExpressionLanguage();
```

Comment:
With that argument I would consider not making it injectable at all. We also control the bundle which we can update to inject this without bothering the developer so I'm guessing we're good? :)

As for others, the example is quite clear so copying code from there shouldn't be too hard.